custom_metrics/wpt_bodies.js

```diff
@@ -0,0 +1,1366 @@
+          else 
+            item.http++;
+
+          if (www) 
+            item.www++;   
+
+          if (jsonld) 
+            item.jsonld++;
+          else 
+            item.microdata++;
+
+          if (!target.context_hostnames.includes(hostname)) {
+            target.context_hostnames.push(hostname);
+          }
+
+          if (name === "schema.org/SearchAction") {
+            target.sitelinks_search_box = true;
+          }
+        }
+
+        function gatherStructuredData(d) {
+
+          let target = {
+            jsonld_and_microdata_types: [],
+            jsonldIds: 0,
+            jsonldReferencedIds: 0,
+            microdataIds: 0,
+            microdataReferencedIds: 0,
+            jsonlsMicrodataCommonIds: 0,
+            logo: false,
+            sitelinks_search_box: false,
+            same_as_values: [],
+            items_by_format: {
+              microformats2: 0,
+              microdata: 0,
+              jsonld: 0,
+              rdfa: 0
+            },
+            context_hostnames: []
+          };
+
+          let jsonldIds = {};
+
+          // json-ld
+          let jsonld_scripts = Array.from(d.querySelectorAll('script[type="application/ld+json"]'));
+
+          target.jsonld_scripts = {
+            count: jsonld_scripts.length,
+            errors: jsonld_scripts.filter(e => {
+              try {
+                var cleanText = e.textContent.trim();
+                var cleanText = cleanText.replace(/^\/\*(.*?)\*\//g, ''); // remove * comment from start (could be for CDATA section) does not deal with multi line comments
+                var cleanText = cleanText.replace(/\/\*(.*?)\*\/$/g, ''); // remove * comment from end (could be for CDATA section) does not deal with multi line comments
+
+                nestedJsonldLookup(target, jsonldIds, JSON.parse(cleanText), 0, "http://no-context.com/");
+                return false; // its good
+              }
+              catch(e) {
+                return true; // its bad
+              }
+            }).length
+          };
+
+          // microdata
+          let microdataNodes = d.querySelectorAll('[itemtype]');
+          for (let i = 0, len = microdataNodes.length; i < len; i++) {
+            let node = microdataNodes[i];
+
+            link.href = node.getAttribute('itemtype');
+
+            let type = link.href;
+
+            addType(target, target.jsonld_and_microdata_types, type, false);  
+
+            target.items_by_format.microdata++;    
+          }
+
+          if (d.querySelector("[itemprop$='logo']")){
+            target.logo = true;
+          }
+          
+          if (d.querySelector("[itemtype$='SearchAction']")){
+            target.sitelinks_search_box = true;
+          }
+
+          let microdataIds = {};
+
+          let microdataItemIdNodes = [...d.querySelectorAll('[itemid]')];
+
+          microdataItemIdNodes.forEach((n) => {
+            link.href = n.getAttribute('itemid');
```

Comment:
The whitespace here is weird

Reply:
I was thinking about it not being a normal space character. Then got it. The lines after are tabbed in one more level for some reason. Fixed.